velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -271,9 +274,15 @@ RowVectorPtr NestedLoopJoinProbe::generateOutput() {
   return std::move(output_);
 }
 
-bool NestedLoopJoinProbe::advanceProbeRow() {
+bool NestedLoopJoinProbe::advanceProbe() {
   if (hasProbedAllBuildData()) {
-    ++probeRow_;
+    // For cross joins, if there is a single record on the build side, we return
+    // batches containing all probe records from `input_` at a time.
+    if (isCrossJoin() && buildRowCount_ == 1) {
```

Comment:
How easy is it to handle the case `buildVectors_.size() == 1` differently (you only need to add a branch in `advanceProbe` and `getNextCrossProductBatch`, no)?  With these 2 cases covered we should cover most regular usage of NLJ.  Otherwise I am afraid we need another diff to address this very soon.

Reply:
Discussed offline, we will have subsequent changes to fix this immediately.

Reply:
Yep, I'll add this in a follow up PR soon. Just trying to keep them a bit smaller and more readable.